lib/matplotlib/tests/test_axes.py

```diff
@@ -2212,6 +2212,8 @@ def test_scatter_color(self):
             plt.scatter([1, 2], [1, 2], color=[0.1, 0.2])
         with pytest.raises(ValueError):
             plt.scatter([1, 2, 3], [1, 2, 3], color=[1, 2, 3])
+        with pytest.raises(ValueError):
+            plt.scatter([1, 2], [1, 2], color='blue\n')
```

Comment:
This raised a value error before, so maybe test the error message with a `matches=...` to the `pytest.raises`?

Reply:
Isn't this just to test whether an error gets raised when given the ``blue\n`` argument? I know that the error was thrown before but in addition to changing the error message slightly, we wanted to add a test.
Maybe, I am not understanding you correctly...

Reply:
I think we also want to test that the repr is being used, not the str...

Reply:
Alright, now I get it...I just updated it, thanks. Another point on that notice: The other examples use the ``color`` parameter, not the ``c`` parameter. Isn't that wrong? I now updated mine to ``c`` since the error message for ``color`` is totally different.

Reply:
Okay, now I got what you mean. I changed it but I now realise that this is misplaced here. It now reads

```python
with pytest.raises(ValueError, match=f"'c' argument must be a color, "
                           f"a sequence of colors, or a sequence of numbers, "
                           f"not {c!r}"):
            plt.scatter([1, 2], [1, 2], color='blue\n')
```

But the entire section is about examples, where ``color`` was used and ``c`` should have been used. However, this is not the case here. This case is about ``c`` already being used but the error message being too unspecific. The example should read instead

```python
with pytest.raises(ValueError, match=f"'c' argument must be a color, "
                           f"a sequence of colors, or a sequence of numbers, "
                           f"not {c!r}"):
            plt.scatter([1, 2], [1, 2], c='blue\n')
```

but be placed in a different section, maybe underneath a comment like this:

# Check for the specificity of error message if invalid 'c' argument is given

Since I don't know if such a section already exists and - if not - where to place it, I wanted to wait for your opinion.

Reply:
I would go with a slightly simpler test and only search for a literal new line (''\n'") rather than trying to match the whole thing.  

The input here is also a regular expression which can require escaping a lot of symbols which can get tedious.   I am also not sure we want to pin down in a test the exact _wording_ just that we got back the reper of the string you put in.

Reply:
Okay, I understand. Nevertheless, the question where to put it remains. Should I just append the example to the previous ones separated by a comment?

Reply:
It is not super important either way.  The benefit of putting it in an existing test is we have fewer tests and the test suite runs marginally faster, the benefit of putting it in its own test is if it fails it is much more obvious what part of the test is failing!